### PR TITLE
Clarified stac_extensions

### PR DESCRIPTION
**Pull request by m-mohr**
**Related Issue(s):** #1077

**Proposed Changes:**

1. For context see #1077, but it basically makes it more consistent that stac_extensions also allow to list extensions that are used in collection summaries.
2. Re-phrased the stac_extensions description in all specs.
3. Removed the wording around "multiple affected parts" as that should be clear from each individual stac_extensions description, which requires to list stac_extensions if they are implemented. That was mostly for the commons extension back in the days, but nowadays had no real good example anymore. The data cube example in collections for example was not really correct.

**PR Checklist:**

- [x] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [ ] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [ ] This PR affects the [STAC API spec](https://github.com/radiantearth/stac-api-spec), and I have opened issue/PR #XXX to track the change.

### CODE REVIEW

**Review by cholmes — APPROVED** [review]

File: collection-spec/collection-spec.md [hunk 1]
```diff
@@ -65,16 +65,11 @@ In general, STAC versions can be mixed, but please keep the [recommended best pr
...
-If an extension has influence on multiple parts 
-of the whole STAC structure, it must be listed in all affected parts (e.g. Collection and Item for the `datacube` extension). 
-If a structure, such as the summaries extension, provides fields in their JSON structure, these extensions must not be listed 
-here as they don't extend the Collection itself. For example, if a Collection includes the field `sat:platform` in the 
-summaries, the Collection should not list the `sat` extension in the `stac_extensions` field.
+A list of extensions the Collection implements.
+The list consists of URLs to JSON Schema files that can be used for validation.
+This list must only contain extensions that extend the Collection specification itself,
+see the the 'Scope' for each of the extensions.
+This must **not** declare the extensions that are only implemented in child Collection or Items.
```
Comment (philvarner) [inline collection-spec/collection-spec.md:72]:
```suggestion
This must **not** declare the extensions that are only implemented in child Collection or Item objects.
```

File: item-spec/item-spec.md [hunk 2]
```diff
@@ -73,11 +73,10 @@ In general, STAC versions can be mixed, but please keep the [recommended best pr
 
 #### stac_extensions
 
-A list of extensions the Item implements.  
-This list must only contain extensions that extend the Item itself, see the the 'Scope' column in the list of 
-extensions. The list contains URLs to the JSON Schema files it can be validated against.
-If an extension such as the `tiled-assets` extension has influence on multiple parts of the whole catalog 
-structure, it must be listed in all affected parts (e.g. Catalog, Collection and Item for the `tiled-assets` extension).
+A list of extensions the Item implements.
```
Comment (philvarner) [inline item-spec/item-spec.md:76]: thanks for rewriting this! I updated this yesterday and only removed the outdated text about shortnames. This sounds much better.

File: catalog-spec/catalog-spec.md [hunk 3]
```diff
@@ -61,9 +61,10 @@ In general, STAC versions can be mixed, but please keep the [recommended best pr
 #### stac_extensions
 
 A list of extensions the Catalog implements.
-This list must only contain extensions that extend the Catalog itself, see the the 'Scope' column in the list of
-extensions. This does NOT declare the extensions of child Catalog, Collection, or Item
-objects. The list contains URLs to the JSON Schema files it can be validated against.
+The list consists of URLs to JSON Schema files that can be used for validation.
+This list must only contain extensions that extend the Catalog specification itself,
+see the the 'Scope' for each of the extensions.
+This must **not** declare the extensions that are only implemented in child Collection or Items.
```
[the remark this answers is not in the document]
Comment (m-mohr) [reply catalog-spec/catalog-spec.md:67]: And I guess it should also be plural for Collection?!

```suggestion
This must **not** declare the extensions that are only implemented in child Collections or Items.
```

Comment (philvarner) [inline catalog-spec/catalog-spec.md:67]:
```suggestion
This must **not** declare the extensions that are only implemented in child Collection objects or child Item objects.
```

[the remark this answers is not in the document]
Comment (philvarner) [reply catalog-spec/catalog-spec.md:67]: Ah, good point. This is ambiguous in English -- I think it's best just to make it explicit. I just submitted another change to do that.

File: collection-spec/collection-spec.md (hunk 1 above)
Comment (philvarner) [inline collection-spec/collection-spec.md:72]:
```suggestion
This must **not** declare the extensions that are only implemented in child Collection objects or child Item objects.
```

**Comment by emmanuelmathot** [comment]
Shouldn't we add in the collection spec a clarification that explains that `stac_extensions` in collection should **not** declare extensions that are only summarized as in example collection https://github.com/radiantearth/stac-spec/blob/master/examples/collection-only/collection.json

**Comment by m-mohr** [comment]
Good point @emmanuelmathot, although I'm not sure whether we should add extensions to stac_extensions that are summarized in colelctions or not. I think best would be to add them, but then we need to change examples.

**Review by matthewhanson — APPROVED** [review]

**Review by philvarner — APPROVED** [review]

**Comment by m-mohr** [comment]
DON'T MERGE YET! 

We should still discuss, whether summaries should have an influence on stac_extensions or not.

**Comment by m-mohr** [comment]
Agreed on it for future-proofness etc.

**Pull request merged**